frame/evm/src/lib.rs

```diff
@@ -211,6 +211,17 @@ impl<H: Hasher<Out=H256>> AddressMapping<AccountId32> for HashedAddressMapping<H
 	}
 }
 
+pub trait BlockHashMapping {
+	fn block_hash(number: u32) -> H256;
+}
+
+impl<T: Config> BlockHashMapping for Module<T> {
+	fn block_hash(number: u32) -> H256 {
+		let number = T::BlockNumber::from(number);
+		H256::from_slice(frame_system::Module::<T>::block_hash(number).as_ref())
+	}
+}
+
```

Comment:
I wonder if we should create a struct specifically for this impl. The reason is that it would make the config trait read more nicely in the runtime's `lib.rs`.

```rust
/// Makes the Substrate-style block hashes from FRAME System available to the EVM
pub struct SubstrateBlockHashMapping;
```

Reply:
Yeah I agree might be more readable but the problem I see with that is once again we include stuff on the template runtime, which is not really part of Frontier and must be copy pasted.

By implementing it in EVM and Ethereum is already built-in and covers all the functionality that is needed for projects that integrate pallet-ethereum and projects that depend on pallet-evm alone. I don't see the case when one might want to implement it in a different way and, if that is the case, people can still do it.

Reply:
> Yeah I agree might be more readable but the problem I see with that is once again we include stuff on the template runtime, which is not really part of Frontier and must be copy pasted.

Why does it introduce additional stuff in template? Can't we just implement `SubstrateBlockHashMapping` in `pallet-evm` and import it to the template?

Reply:
Misunderstanding on my part. @JoshOrndorff I thought you were suggesting defining and implementing the struct on the runtime.